plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/UnregisterTableProcedure.java

```diff
@@ -0,0 +1,113 @@
+import io.trino.spi.connector.ConnectorAccessControl;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.SchemaNotFoundException;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.TableNotFoundException;
+import io.trino.spi.procedure.Procedure;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import java.lang.invoke.MethodHandle;
+
+import static io.trino.plugin.base.util.Procedures.checkProcedureArgument;
+import static io.trino.spi.StandardErrorCode.UNSUPPORTED_TABLE_TYPE;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.lang.String.format;
+import static java.lang.invoke.MethodHandles.lookup;
+import static java.util.Objects.requireNonNull;
+
+public class UnregisterTableProcedure
+        implements Provider<Procedure>
+{
+    private static final MethodHandle UNREGISTER_TABLE;
+    private static final String SYSTEM_SCHEMA = "system";
+    private static final String PROCEDURE_NAME = "unregister_table";
+    private static final String SCHEMA_NAME = "SCHEMA_NAME";
+    private static final String TABLE_NAME = "TABLE_NAME";
+
+    static {
+        try {
+            UNREGISTER_TABLE = lookup().unreflect(UnregisterTableProcedure.class.getMethod("unregisterTable", ConnectorAccessControl.class, ConnectorSession.class, String.class, String.class));
+        }
+        catch (ReflectiveOperationException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    private final TransactionalMetadataFactory hiveMetadataFactory;
+
+    @Inject
+    public UnregisterTableProcedure(TransactionalMetadataFactory hiveMetadataFactory)
+    {
+        this.hiveMetadataFactory = requireNonNull(hiveMetadataFactory, "hiveMetadataFactory is null");
+    }
+
+    @Override
+    public Procedure get()
+    {
+        return new Procedure(
+                SYSTEM_SCHEMA,
+                PROCEDURE_NAME,
+                ImmutableList.of(
+                        new Procedure.Argument(SCHEMA_NAME, VARCHAR),
+                        new Procedure.Argument(TABLE_NAME, VARCHAR)),
+                UNREGISTER_TABLE.bindTo(this));
+    }
+
+    public void unregisterTable(ConnectorAccessControl accessControl, ConnectorSession session, String schemaName, String tableName)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            doUnregisterTable(accessControl, session, schemaName, tableName);
+        }
+    }
+
+    private void doUnregisterTable(ConnectorAccessControl accessControl, ConnectorSession session, String schemaName, String tableName)
+    {
+        checkProcedureArgument(schemaName != null, "schema_name cannot be null");
+        checkProcedureArgument(tableName != null, "table_name cannot be null");
+        SchemaTableName schemaTableName = new SchemaTableName(schemaName, tableName);
+
+        accessControl.checkCanDropTable(null, schemaTableName);
```

Comment:
Add a test to verify the effectiveness of the access control.

See for a PoC https://github.com/trinodb/trino/blob/fd14f9f477c48bb3d14dc3480715c792c2246617/plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java#L6098-L6135